src/connectors/components/HeaderMenuRequestConnector.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+import { connect } from 'react-redux'
+
+import { dispatchActionOnly } from '../../actions/indexActions'
+import * as Constants from '../../constants/indexConstants'
+import s from '../../locales/strings.js'
+import type { Dispatch, State } from '../../modules/ReduxTypes.js'
+import { openHelpModal } from '../../modules/UI/components/HelpModal/actions'
+import {MenuDropDown} from '../../modules/UI/components/MenuDropDown/MenuDropDown.ui.js'
+import * as Styles from '../../styles/indexStyles'
+import THEME from '../../theme/variables/airbitz'
+
+export const mapStateToProps = (state: State) => {
+  const useLegacyAddress = state.ui.scenes.requestType.useLegacyAddress
+  const uniqueLegacyAddress = state.ui.scenes.requestType.uniqueLegacyAddress
+
+  const addressToggle = {
+    label: useLegacyAddress ? s.strings.title_use_regular_address : s.strings.title_use_legacy_address, // tie into,
+    key: useLegacyAddress ? s.strings.title_use_regular_address : s.strings.title_use_legacy_address,
+    value: {
+      title: useLegacyAddress ? s.strings.title_use_regular_address : s.strings.title_use_legacy_address
+    }
+  }
+  const help = {
+    label: s.strings.string_help,
+    key: s.strings.string_help,
+    value: {
+      title: Constants.HELP_VALUE
+    }
+  }
+  const data = uniqueLegacyAddress ? [addressToggle, help] : [help]
+  return {
+    style: {
```

Comment:
please do not create new objects in mapStateToProps, this prevents React from skipping renders if the props don't change

Reply:
Oh, good catch , I am actually going to fix in HeaderExchangeConnector.js, this is where some of this code was copied from when I started this connector.